core/osrd-reporting/src/main/java/fr/sncf/osrd/reporting/exceptions/OSRDError.java

```diff
@@ -25,25 +30,27 @@
  * </pre>
  */
 @SuppressWarnings("serial") // Removes the warning about List not being serializable (moshi doesn't support ArrayList)
-public abstract class OSRDError extends RuntimeException {
+public class OSRDError extends RuntimeException {
 
     @Serial
     private static final long serialVersionUID = 1197516372515951853L;
 
-    public static final String osrdErrorType = "core";
+    public static final transient String osrdErrorType = "core";
 
     /** Detailed error message */
-    public final String message;
+    public transient String message;
 
     /** Stack trace of context */
     public final List<ErrorContext> trace = new ArrayList<>();
 
     /** Whether this is an internal or user error, used to determine if we send a 400 or 500 code */
     public final ErrorCause cause;
```

Comment:
We could adapt OSRDError to match the serialized type, meaning:

- Three fields: `type`, `message` and `context`.
- The context can contain a `trace` by default.

By doing so we don't have to implement a custom moshi adapter to serialize our errors. 
We don't have to serialize the `cause` (it's only here to define the response code).